mesonbuild/backend/ninjabackend.py

```diff
@@ -2734,10 +2734,14 @@ def generate_llvm_ir_compile(self, target, src):
         return (rel_obj, rel_src)
 
     @lru_cache(maxsize=None)
-    def generate_inc_dir(self, compiler: 'Compiler', d: str, basedir: str, is_system: bool) -> \
+    def generate_inc_dir(self, compiler: 'Compiler', d: str, basedir: T.Optional[str], is_system: bool) -> \
             T.Tuple['ImmutableListProtocol[str]', 'ImmutableListProtocol[str]']:
+        # If the basedir is None, this is a system provided path
+        if basedir is None:
+            return (compiler.get_include_args(d, is_system), [])
+
         # Avoid superfluous '/.' at the end of paths when d is '.'
-        if d not in ('', '.'):
+        if d not in {'', '.'}:
```

Comment:
Is this a candidate for frozenset?

Reply:
Python will optimize an anonymous set like this into a frozenset when it byte compiles the .py files

Reply:
Also there is no speed difference between sets and frozensets AFAIK -- the difference is that the former has mutating methods and the latter can be hashed / used as dictionary keys.

There is, generally, no real purpose in using frozenset() instead of set() except inasmuch as you care about asserting immutability.

Using set *literals* is useful because it can be compiled in bytecode, but that isn't conditional on its being optimized into specifically a frozenset. It just happens to be an internal implementation detail of python.